### ISSUE TITLE
Allow blacklisting user across the network

### ISSUE THREAD

**Issue by CoconutMacaroon** [Charcoal-SE/SmokeDetector#6845]
I'm cocomac on Stack Exchange. [Here is my Ask Ubuntu account](https://askubuntu.com/users/1438484/cocomac), which has GitHub linked on it. 

### Is your feature request related to a problem? Please describe.
I'd like the ability to be able to blacklist a SE user, and have all of their SE per-site accounts get blacklisted as well. There aren't any cases that I know of where a user spams on one site, and is a great user on a different one. Most of the time, if they are spamming/trolling/being rude or abusive, they really should get blacklisted everywhere. I ran into this in recently, where I had to blacklist each of their per-site accounts manually after they were causing trouble on several sites.
### Describe the solution you'd like
Having one command to blacklist all of a user's SE accounts (and any that they create in the future, ideally) would be very helpful.
### Describe alternatives you've considered
Blacklist the user on each site by hand - slow, tedious
### Additional context
There was a conversation in Charcoal HQ:

https://chat.stackexchange.com/transcript/message/60711269#60711269

**Comment by CoconutMacaroon** [comment]
Hey! I'm still interested in this. I have no idea how to do it (maybe I can figure it out), but it is still important to me

**Comment by makyen** [comment]
A prerequisite for this is to get the data from the SE API, which is also a prerequisite for #6695, which is also tied in to #4145 (which required writing [this](https://stackapps.com/questions/9259/how-do-i-see-the-differences-between-two-or-more-se-api-filters-and-merge-into-a), and which was derived from #4074). The changes for getting the data from the SE API are also somewhat tied into the `Post` changes, but are not hard-coupled.

In other words, I have code that does some of the prerequisites for this, which I need to split out of a larger set of changes and get merged.

**Comment by makyen** [comment]
We will also need to look at how to handle when the blacklist request is for a user who has a notable amount of reputation on any particular site, which might not be the site for which they are being blacklisted. Such things *do* happen, both intentionally and by mistake. It also is possible where the user should be blacklisted on one site, but not all sites.

Yes, actual straight spammers rarely have good posts on other sites, but that's not all the cases where blacklisting actually gets used (even if the situation is not one where we want user blacklisting to be used).

In other words, we need to be sure that it's possible to blacklist a user on just a single site and that it's possible to blacklist the user based on their SE account, thus blacklisting them on all sites where they have, or later create, a profile. At this point, it's reasonable to just have a single entry for the user's SE account in the blacklist, rather than needing to blacklist each current site profile and then track when they create new profiles.

**Comment by makyen** [comment]
When we do this, we should also allow whitelisting a user by their SE account.

**Comment by gparyani** [comment]
There's another good reason for this: see [this moderator-only link](https://chat.stackexchange.com/transcript/118161?m=65169781#65169781).

**Comment by makyen** [comment]
The SE API requests for questions, answers, and posts request the user's SE account ID as of the merging of PR #7395. As I recall, that data is not, yet, exposed in the `Post` object, but we receive it.

**Comment by CoconutMacaroon** [comment]
Rough proposal:
- [ ] Make `!!/isblu` check the network blacklist and the per-site blacklist, and return information about on which (if any) of those blacklists a given user is on. I'd probably go with a slightly-modified version of [the phrasing Makyen suggested](https://github.com/Charcoal-SE/SmokeDetector/commit/2db2bccab9b24d49a609e1c3518a87b6c417a4af#r94184039).
- [ ] Create `isblu-site` and `isblu-net` that only check the site blacklist and the network-wide one, respectively.
- [ ] Add two new commands, `!!/addblu-net` and `!!/rmblu-net` to add or remove (respectively) users from the network-wide blacklist. This would be a manual decision made by a human, `tpu` would be left as-is.
- [ ] Leave `!!/addblu` mostly as-is, but issue a warning if the user being blacklisted is already on the network-wide blacklist.
